internal/pkg/exec/docker.go

```diff
@@ -166,6 +166,36 @@ func (c DockerCommand) CheckDockerEngineRunning() error {
 	}
 }
 
+// GetPlatform will run the `docker version` command to get the OS/Arch.
+func (c DockerCommand) GetPlatform() (operatingSystem, architecture string, err error) {
+	var os string
+	var arch string
+	if _, err := exec.LookPath("docker"); err != nil {
+		return "", "", ErrDockerCommandNotFound
+	}
+	osBuf := &bytes.Buffer{}
+	err = c.runner.Run("docker", []string{"version", "-f", "'{{.Server.Os}}'"}, Stdout(osBuf))
+	if err != nil {
+		return "", "", fmt.Errorf("get docker os: %w", err)
+	}
+	if c.buf != nil {
+		osBuf = c.buf
+	}
+	os = strings.TrimSuffix(strings.TrimPrefix(strings.TrimSpace(osBuf.String()), "'"), "'")
+
+	archBuf := &bytes.Buffer{}
+	err = c.runner.Run("docker", []string{"version", "-f", "'{{.Server.Arch}}'"}, Stdout(archBuf))
```

Comment:
An alternative to issuing two commands:
```go
// Use the "json" function
buf := new(bytes.Buffer)
c.runner.Run("docker", []string{"version", "-f", "'{{json .Server}}'"}, Stdout(buf))

// unmarshal into data
daemon := struct {
  OS string `json:"Os"`
  Arch string `json:"Arch"`
}{}
json.Unmarshal(buf.Bytes(), &daemon)
```

Reply:
that's what I had originally...will switch it back!